Allow downloading of all columns in the NASA Exoplanet Archive

mwcraig opened the pull request:
This PR fixes #1027 by adding an option to get all columns (instead of just the default set) from the NASA Exoplanet Archive.

One test will fail; if I download with `all_columns=True` first, then `False` fails. I assume there is some cacheing going on that I haven't been able to stop.

Arguably it would make more sense just to make it *always* return all columns (though @bmorris3 may know of a reason not too); then we would only need one test :).

This catalog is more complete than than the Exoplanet Orbit Database, so it would be nice to have it return more transit-related parameters.

bsipocz commented on astroquery/nasa_exoplanet_archive/nasa_exoplanet_archive.py:
```diff
@@ -104,6 +112,9 @@ def query_planet(self, planet_name, table_path=None):
             download of the table from the internet.
+        all_columns : bool (optional)
+            Return all available columns. The default returns only the
+            columns in the default category at the link above.
```
could you add an extra empty line here, the docstring doesn't render correctly atm.

bsipocz commented on astroquery/nasa_exoplanet_archive/nasa_exoplanet_archive.py:
```diff
@@ -58,14 +58,21 @@ def get_confirmed_planets_table(self, cache=True, show_progress=True,
+        all_columns : bool (optional)
+            Return all available columns. The default returns only the
+            columns in the default category at the link above.
         Returns
```
could you add an extra empty line here, the docstring doesn't render correctly atm.

bsipocz commented:
@mwcraig - Could you also add a changelog entry?

mwcraig commented:
@bsipocz -- thanks, have addressed the comments. How do recommend proceeding regarding the failing test? Looks like I either need to clear the cache between tests or switch to always returning all columns....

bsipocz commented:
@mwcraig - There is the `cache` kwarg is ignored then it's a bug on its own right. Could you open an issue for it please?. But it's beyond this PR, so feel free to move your new test to the bottom of the tests, then arguably everything should still pass.

mwcraig commented:
Found and fixed the cacheing error. The class was doing its own cacheing by checking whether `self._table` existed; since `NasaExoplanetArchive` is an instance of the class, `self._table` is set the first time `get_confirmed_planets_table` is called and stays the same for all tests, so the first test was setting the table contents and ignoring the cache setting.

bsipocz commented:
Thanks @mwcraig!
